test/workbox-strategies/sw/test-NetworkFirst.mjs

```diff
@@ -141,12 +141,37 @@ describe(`NetworkFirst`, function() {
       const cache = await caches.open(cacheNames.getRuntimeName());
       await cache.put(request, injectedResponse.clone());
 
-      const handlePromise = networkFirst.handle({
+      const [handlePromise, donePromise] = networkFirst.handleAll({
+        request,
+        event,
+      });
+
+      await eventDoneWaiting(event);
+      await donePromise;
+
+      const populatedCacheResponse = await handlePromise;
+      await compareResponses(populatedCacheResponse, injectedResponse, true);
+    });
+
+    it(`should signal completion if the network request completes before timing out`, async function() {
+      const request = new Request('http://example.io/test/');
+      const event = new FetchEvent('fetch', {request});
+      spyOnEvent(event);
+
+      const networkTimeoutSeconds = 1000;
+
+      const injectedResponse = new Response('response body');
+      sandbox.stub(self, 'fetch').resolves(injectedResponse);
+
+      const networkFirst = new NetworkFirst({networkTimeoutSeconds});
+
+      const [handlePromise, donePromise] = networkFirst.handleAll({
         request,
         event,
       });
 
       await eventDoneWaiting(event);
+      await donePromise;
```

Comment:
I think you also want to test that the handler completes before 1000ms passes, right? I.e. from my read of your changes this test would still pass with the old logic.

Reply:
Prior to my changes, donePromise never resolved, instead of resolving after 1000ms.

But verifying that it completes quickly is a good addition; thank you. Please see the latest commit.

(I also changed the networkTimeoutSeconds from 1000 to 10; I thought that might avoid potential confusion between 1000 seconds and 1000 ms, and 10 seconds is still plenty to make the test serve its purpose of failing if the promise doesn't resolve.)

Reply:
Same comment as above, just remove the call to `eventDoneWaiting()`.

Let me know if for some reason the tests don't pass with that change (it shouldn't have any effect).